sklearn/ensemble/_iforest.py

```diff
@@ -402,6 +445,22 @@ def decision_function(self, X):
             The anomaly score of the input samples.
             The lower, the more abnormal. Negative scores represent outliers,
             positive scores represent inliers.
+
+        Notes
+        -----
+        The decision_function method can be parallelized by setting a joblib context.
+        This inherently does NOT use the ``n_jobs`` parameter initialized in the class,
+        which is used during ``fit``. This is because, calculating the score may
+        actually be faster without parallelization for a small number of samples.
+        The user can set the number of jobs in the joblib context to control the
+        number of parallel jobs.
+
+        .. code-block:: python
+
+            from joblib import parallel_backend
+
+            with parallel_backend('loky', n_jobs=4):
```

Comment:
Does the loky backend work with the `require="sharedmem"` ? I recall `loky` is a "better multiprocessing" and  `_parallel_compute_tree_depths` is using a `threading.Lock`, which is a lock for multi-threading.

Reply:
Technically loky backend does not, but the docs on Parallel sounds like it handles that:

```
require: ‘sharedmem’ or None, default=None
Hard constraint to select the backend. If set to ‘sharedmem’, the selected backend will be single-host and thread-based even if the user asked for a non-thread based backend with [parallel_config()](https://joblib.readthedocs.io/en/latest/generated/joblib.parallel_config.html#joblib.parallel_config).
```
https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html

I tested it out and the results were the same, but I think the docstring should say `'threading', so I changed that.

Reply:
For reference, see: [cd185ca](https://github.com/scikit-learn/scikit-learn/pull/28622/commits/cd185cabe143b47ae421cfa4f00ef5f19d02bdaa)